src/Thermochimica-cxx.C

```diff
@@ -0,0 +1,302 @@
+#include "Thermochimica.h"
+
+namespace Thermochimica
+{
+
+  void thermochimica()
+  {
+    TCAPI_thermochimica();
+  }
+
+  void setThermoFilename(const std::string &filename)
+  {
+    TCAPI_setThermoFilename(filename.c_str(), filename.length());
+  }
+
+  void setUnitTemperature(const std::string &tunit)
+  {
+    TCAPI_setUnitTemperature(tunit.c_str(), tunit.length());
+  }
+
+  void setUnitPressure(const std::string &punit)
+  {
+    TCAPI_setUnitPressure(punit.c_str(), punit.length());
+  }
+
+  void setUnitMass(const std::string &munit)
+  {
+    TCAPI_setUnitMass(munit.c_str(), munit.length());
+  }
+
+  void setUnits(const std::string &tunit, const std::string &punit, const std::string &munit)
+  {
+    setUnitTemperature(tunit);
+    setUnitPressure(punit);
+    setUnitMass(munit);
+  }
+
+  void setStandardUnits()
+  {
+    TCAPI_setStandardUnits();
+  }
+
+  void setModelicaUnits()
+  {
+    TCAPI_setModelicaUnits();
+  }
+
+  void setTemperaturePressure(double temperature, double pressure)
+  {
+    TCAPI_setTemperaturePressure(&temperature, &pressure);
+  }
+
+  void setElementMass(int element, double mass)
+  {
+    TCAPI_setElementMass(&element, &mass);
+  }
+
+  int checkInfoThermo()
+  {
+    int idbg;
+    TCAPI_checkInfoThermo(&idbg);
+    return idbg;
+  }
+
+  void parseThermoFile()
+  {
+    TCAPI_sSParseCSDataFile();
+  }
+
+  void setPrintResultsMode(int mode)
+  {
+    TCAPI_setPrintResultsMode(&mode);
+  }
+
+  void printResults()
+  {
+    TCAPI_printResults();
+  }
+
+  void resetInfoThermo()
+  {
+    TCAPI_resetInfoThermo();
+  }
+
+  void resetThermo()
+  {
+    TCAPI_resetThermo();
+  }
+
+  void resetThermoAll()
+  {
+    TCAPI_resetThermoAll();
+  }
+
+  // re-initialization-related functions
+  void saveReinitData()
+  {
+    TCAPI_saveReinitData();
+  }
+
+  std::pair<int, int> getReinitDataSizes()
+  {
+    int elements, species;
+    TCAPI_getReinitDataSizes(&elements, &species);
+    return {elements, species};
+  }
+
+  double getMolesPhase()
+  {
+    double molesPhase;
+    TCAPI_getMolesPhase(&molesPhase);
+    return molesPhase;
+  }
+
+  std::vector<int> getAssemblage()
+  {
+    std::pair<int, int> p = getReinitDataSizes();
+    int assemblageArray[p.first];
+
+    TCAPI_getAssemblage(assemblageArray);
+
+    std::vector<int> assemblage(assemblageArray, assemblageArray + sizeof assemblageArray / sizeof assemblageArray[0]);
+
+    return assemblage;
+  }
+
+  void setReinitRequested(bool requested)
+  {
+    int req = (requested) ? req = 1 : 0;
+    TCAPI_setReinitRequested(&req);
+  }
+
+  void resetReinit()
+  {
+    TCAPI_resetReinit();
+  }
+
+  std::vector<double> getAllElementPotential()
+  {
+    std::pair<int, int> p = getReinitDataSizes();
+    double potentialArray[p.first];
+
+    TCAPI_getAllElementPotential(potentialArray);
+
+    std::vector<double> potential(potentialArray, potentialArray + sizeof potentialArray / sizeof potentialArray[0]);
+
+    return potential;
+  }
```

Comment:
```suggestion
  std::vector<double> getAllElementPotential()
  {
    auto [size, idbg] = getReinitDataSizes();
    std::vector<double> potential(size);
    TCAPI_getAllElementPotential(potential.data());
    return potential;
  }
```
maybe also check `idbg` in these routines? or return it?

Reply:
The second index actually isn't a debug flag, it's just another size that we don't need here.